libs/src/api/Users.ts

```diff
@@ -771,6 +993,37 @@ export class Users extends Base {
     }
   }
 
+  async waitForReplicaSetDiscoveryIndexing(
```

Comment:
if i'm understanding this correctly
the `discoveryProvider.getUserReplicaSet({ blockNumber })` call will not retry and 1) return replica set if blocknumber indexed, or 2) throw an error if not indexed

given that CN does not retry on this call, shouldn't there be some retry here until blocknumber is indexed?

also this logic will return replica set if blocknumber indexed, but if the equality check on 1009 returns false then it will keep trying until timeout, when it should actually immediately return?

thats what we had discussed right? i might be misunderstanding the code tho , so lmk

Reply:
Yes, you are correct - I added this same logic in the middleware for fetching the user's replica set and short circuiting after block indexed & not found vs not yet indexed and polling but it should be added here. 
In my local e2e testing it worked, but I must not have testing this part with the delayed indexing.

Reply:
should be addressed in last commit!